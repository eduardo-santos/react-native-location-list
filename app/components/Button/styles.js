import { StyleSheet } from "react-native";

const BORDER_RADIUS = 4;

export default StyleSheet.create({
  container: {
    backgroundColor: "#2695D2",
    paddingVertical: 16,
    paddingHorizontal: 30,
    borderRadius: BORDER_RADIUS,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center"
  },
  button: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center"
  },
  text: {
    color: "#FFFFFF",
    fontSize: 14
  }
});
