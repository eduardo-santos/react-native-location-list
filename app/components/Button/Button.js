import React from "react";
import PropTypes from "prop-types";
import { TouchableHighlight, View, Text } from "react-native";

import styles from "./styles";

const Button = ({
  text,
  onPress,
  width,
  height,
  paddingVertical,
  paddingHorizontal,
  marginTop,
  disabled
}) => {
  const buttonStyles = [styles.container];
  const textStyles = [styles.text];

  if (width) {
    buttonStyles.push({ width });
    buttonStyles.push({ paddingHorizontal: 0 });
  } else if (paddingHorizontal) {
    buttonStyles.push({ paddingHorizontal });
  }

  if (height) {
    buttonStyles.push({ height });
    buttonStyles.push({ paddingVertical: 0 });
  } else if (paddingVertical) {
    buttonStyles.push({ paddingVertical });
  }

  if (marginTop) {
    buttonStyles.push({ marginTop });
  }

  return (
    <TouchableHighlight
      style={buttonStyles}
      onPress={onPress}
      disabled={disabled}
      underlayColor="#2695D2"
    >
      <View style={styles.button}>
        <Text style={textStyles}>{text}</Text>
      </View>
    </TouchableHighlight>
  );
};

Button.propTypes = {
  text: PropTypes.string,
  onPress: PropTypes.func,
  width: PropTypes.any,
  height: PropTypes.any,
  paddingVertical: PropTypes.number,
  paddingHorizontal: PropTypes.number,
  marginTop: PropTypes.number,
  disabled: PropTypes.bool
};

export default Button;
