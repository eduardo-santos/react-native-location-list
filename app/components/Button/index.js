import Button from "./Button";
import styles from "./styles";

export { Button, styles };
