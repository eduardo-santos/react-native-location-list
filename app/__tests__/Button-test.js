import React from "react";

import renderer from "react-test-renderer";
import Button from "../components/Button/Button";

it("renders correctly", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
