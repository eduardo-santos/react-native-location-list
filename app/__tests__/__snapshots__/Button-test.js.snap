// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#2695D2",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingVertical": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
          },
        ]
      }
    />
  </View>
</View>
`;
